File: WinForms/VB/TreeMapMainDemo/frmMain.vb
Imports DevExpress.DXperience.Demos
Imports DevExpress.LookAndFeel
Imports DevExpress.Utils
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraTreeMap.Demos

    Public Class frmMain
        Inherits RibbonMainForm

        Protected Overrides ReadOnly Property DemoName As String
            Get
                Return "WinForms Heatmap, TreeMap & Sunburst"
            End Get
        End Property

        Protected Overrides ReadOnly Property ProductName As String
            Get
                Return "XtraTreeMap"
            End Get
        End Property

        Protected Overrides Sub ShowAbout()
            Call HierarchicalChartControlBase.About()
        End Sub

        Public Overrides ReadOnly Property UseEmptyRootPadding As Boolean
            Get
                Return True
            End Get
        End Property

        Protected Overrides Sub SetFormParam()
            Icon = ResourceImageHelperCore.CreateIconFromResourcesEx("DevExpress.XtraTreeMap.Demos.AppIcon.ico", GetType(frmMain).Assembly)
        End Sub

        Protected Overrides Sub ShowModule(ByVal name As String, ByVal group As GroupControl, ByVal lookAndFeel As DefaultLookAndFeel)
            DemosInfo.DoShowModule(name, group, RibbonMenuManager)
        End Sub

        Protected Overrides ReadOnly Property SupportAdvancedTitlePainting As Boolean
            Get
                Return False
            End Get
        End Property
    End Class
End Namespace

File: WinForms/CS/TreeMapMainDemo/frmMain.cs
using DevExpress.DXperience.Demos;
using DevExpress.LookAndFeel;
using DevExpress.Utils;
using DevExpress.XtraEditors;

namespace DevExpress.XtraTreeMap.Demos {
    public class frmMain : RibbonMainForm {
        protected override string DemoName {
            get { return "WinForms Heatmap, TreeMap & Sunburst"; }
        }
        protected override string ProductName {
            get { return "XtraTreeMap"; }
        }
        protected override void ShowAbout() {
            TreeMapControl.About();
        }
        public override bool UseEmptyRootPadding {
            get { return true; }
        }
        protected override void SetFormParam() {
            Icon = ResourceImageHelper.CreateIconFromResourcesEx("DevExpress.XtraTreeMap.Demos.AppIcon.ico", typeof(frmMain).Assembly);
        }
        protected override void ShowModule(string name, GroupControl group, DefaultLookAndFeel lookAndFeel) {
            DemosInfo.DoShowModule(name, group, RibbonMenuManager);
        }
        protected override bool SupportAdvancedTitlePainting {
            get { return false; }
        }
    }
}
